homeassistant/components/unifi/sensor.py

```diff
@@ -185,3 +236,172 @@ async def options_updated(self) -> None:
         """Config entry options are updated, remove entity if option is disabled."""
         if not self.controller.option_allow_uptime_sensors:
             await self.remove_item({self.client.mac})
+
+
+@dataclass
+class UniFiDeviceSensorEntityDescription(SensorEntityDescription):
+    """Describes UniFi device sensor entities."""
+
+    enabled: str | None = None
+    value_fn: Callable[[UniFiDevice], Any] | None = None
+
+    def get_value(self, device: UniFiDevice) -> Any:
+        """Return value from UniFi device."""
+        if self.value_fn is not None:
+            return self.value_fn(device)
+
+        return None
+
+    def is_enabled(self, device: UniFiDevice) -> bool:
+        """Return whether the entity should be enabled."""
+        if self.enabled is None:
+            return True
+
+        return device.raw.get(self.enabled, False)
+
+
+DEVICE_SENSORS: tuple[UniFiDeviceSensorEntityDescription, ...] = (
+    UniFiDeviceSensorEntityDescription(
+        key=CPU_UTILIZATION_SENSOR,
+        name="CPU Utilization",
+        native_unit_of_measurement=PERCENTAGE,
+        icon="mdi:speedometer",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        enabled="system-stats",
+        value_fn=lambda device: float(device.raw.get("system-stats", {}).get("cpu")),
+    ),
+    UniFiDeviceSensorEntityDescription(
+        key=MEMORY_UTILIZATION_SENSOR,
+        name="Memory Utilization",
+        native_unit_of_measurement=PERCENTAGE,
+        icon="mdi:memory",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        enabled="system-stats",
+        value_fn=lambda device: float(device.raw.get("system-stats", {}).get("mem")),
```

Comment:
Can we expand with explicit system stat attributes in aiounifi?

Reply:
As above, yes I want to do this.